### ISSUE TITLE
I could use a way to read the state of the emulated CPU/memory without holding a mutable reference

### ISSUE THREAD

**Issue by omarandlorraine** [ivanizag/rust-z80#4]
Hi,

thanks for this project! I am planning to use this emulator for an upcoming rewrite of [my project, strop](https://github.com/omarandlorraine/strop).

But to do this I would need to be able to do something like

```
pub struct Z80 {
    pub mach: iz80::PlainMachine,
    pub cpu: iz80::Cpu,
}

fn get_accumulator(&self, e: &Z80) -> u8 {
    e.cpu.registers().a()
}
```

This code fails to compile because
```
error[E0596]: cannot borrow `e.cpu` as mutable, as `e` is not declared as mutable
  --> src/z80/mod.rs:19:9
   |
18 | fn get(&self, e: &Z80) -> u8 {
   |               - help: consider changing this to be mutable: `mut e`
19 |     e.cpu.registers().a()
   |     ^^^^^^^^^^^^^^^^^ cannot borrow as mutable
```

I feel like the best way forward here would be to add a a method to `&Cpu` that returns an immutable reference to the `Registers` and then use its existing methods `a`, `get8`, `pc` etc.

Would you accept a pull request to do this or do you have a preferred way?

**Comment by ivanizag** [comment]
Yes, go ahead please.

**Comment by omarandlorraine** [comment]
Closed with #5, which implements the necessary method,
